### ISSUE TITLE
Schema dump changes primary key from string to integer.

### ISSUE THREAD

**Issue by hectorsq** [rails-sqlserver/activerecord-sqlserver-adapter#373]
Running rake db:schema:dump using 4.0.0 of activerecord-sqlserver-adapter results in the following schema:

```ruby
  create_table "documents", id: false, force: true do |t|
    t.string   "oid",        limit: 32, null: false
    t.string   "code"
    t.string   "name"
    t.string   "event_oid",  limit: 32
    t.datetime "created_at"
    t.datetime "updated_at"
  end
```

Using version 4.1.0 of the gem I get the following schema:

```ruby
  create_table "documents", primary_key: "oid", force: true do |t|
    t.string   "code"
    t.string   "name"
    t.string   "event_oid",  limit: 32
    t.datetime "created_at"
    t.datetime "updated_at"
  end
```

The problem is that if I run rake db:schema:load on a clean database the primary key will be created as ineteger when the original one was a string.

**Comment by metaskills** [comment]
This should be fixed in the latest 4.2 pre-release that just hit.
http://metaskills.net/2015/01/25/activerecord-sqlserver-v4.2.0---code-name-kantishna/

**Comment by guigs** [comment]
I have the same problem, but with 4.0.0. Only before 6e227013a6625388ad8b7c42546939ed39e08f1e db:schema:dump worked with string primary keys.

Unfortunately, I cannot upgrade to 4.2.0 because I'm using Ruby 4.0.x and SQL Server 2008.

Is there a commit that fix this that can be backported to 4.0-stable?

**Comment by metaskills** [comment]
The code in master is big and sweeping and the interfaces for schema compliance is incredible different between 4.0 and 4.2. That is to say, you wont find a single re-usable chunk to transpose to 4.0/4.1 to fix this. Someone is going to have to get in there solve it and to be honest, I am putting all my free time to 4.2. I can say if anyone makes a patch, I will test it out and get it in and update the gem for our older 4-0-stable branch. HTH.
